### ISSUE TITLE
new tab doesn't load custom url, keeps resetting

### ISSUE THREAD

**Issue by TalionDiscord** [cadeyrn/newtaboverride#351]
I have a custom URL set to load google on a new tab (custom URL). when I open a new tab, it doesn't load google anymore, it doesn't load anything. So I go into the settings and where the custom url is typed, I press the enter key so that the text appears underneath that says "Revoke permission Firefox needs your permission to read the news feed from soeren-hentzschel.at. You have already granted this permission. [Revoke permission]".

This fixes it temporarily but then after a bit of time the same thing happens all over again. I think there may be an issue with it remembering the settings. 

This only happened within the past few days and has been working fine for a long time before this issue came up.

**Comment by cadeyrn** [comment]
Which versions of Firefox and New Tab Override do you use?

For a custom URL, you don't need the mentioned permission. It's only needed if you want to show the latest news from that blog on new tabs. But it's interesting that the permission prompt appears when you press enter. This is a bug that I should fix. You don't need to press anything to confirm your input.

If it only happens since a few days, please try to remember what you did, like installing new add-ons, updating something and so on. Please also check if you can reproduce the issue on a new Firefox profile.

**Comment by TalionDiscord** [comment]
I never did anything different with my browser, it just worked the same as it always did up until a few days ago. The version of New Tab Override is 16.0.1 and Firefox version is 128.0.3 (64-bit) Up to date.

However I have just stumbled upon something which may be the issue: When I start a brand new session of Firefox for the first time, and wait for it to load up, and then I open a new tab for the first time in that session, Firefox pops up with a small box asking if I want to keep or remove/discard changes (I have had this before when Firefox crashed and closed itself causing my extentions to take time to reload fully upon opening a new session, and upon opening a new tab it popped up with the same question, I believe it is connected to New Tab Override).

I have just had this pop up now and have clicked "Keep changes". So far it is working normally but if the issue continues I will report it here.

However it is odd that this pop up is occuring in normal sessions, it only ever did this when Firefox crashed and forced closed then reloaded up. I think somewhere there is some sort of issue storing the data the extension needs perhaps? I truly do not know.

**Comment by TalionDiscord** [comment]
Yep, just as I finished that post, I went to open a new tab after clicking "Keep changes" and it refuses to load my custom url. Definitely something not right.

**Comment by cadeyrn** [comment]
> When I start a brand new session of Firefox for the first time, and wait for it to load up, and then I open a new tab for the first time in that session, Firefox pops up with a small box asking if I want to keep or remove/discard changes

This is definitely not normal and could mean that your profile is damaged. Or do you use a pointless "cleaning tool" like CCleaner or something similar?

Please check if you can reproduce the issue on a new Firefox profile.

**Comment by TalionDiscord** [comment]
I don't use any cleaning tools. I started a new session today and it never even gave me the pop-up asking to keep or manage tabs (I found out it was manage tabs. not discard). This time, it just worked on and off, sometimes it will load the custom URL, other times in the same session it won't. It is weird.

So I tried in a new profile like you said, I cannot seem to replicate it, I did have to install the extension again new, so I did that and tried and couldn't replicate the issue in the bit of time I tested it for.

So I switched back to my normal profile, and I have removed the extension, then re-added it as if it were a new extension and set up the custom URL. So far it appears to be working normally but I will let you know if it fails again.
